python/idsse_common/idsse/common/publish_confirm.py

```diff
@@ -175,18 +178,19 @@ def _run(self):
             # Finish closing
             self._connection.ioloop.start()
 
-    def _start(self, callback: Callable[[], None] | None = None):
+    def _start(self, is_ready: Future | None = None):
         """
         Start a thread to handle PublishConfirm operations
 
         Args:
-            callback (Callable[[], None] | None): callback function to be invoked
-                once instance is ready to publish messages (all RabbitMQ connection and channel
-                are set up, delivery confirmation is enabled, etc.). Default to None.
+            is_ready (Future | None): optional Python Future that will be resolved once
+                PublishConfirm is ready to publish messages (all RabbitMQ connection and channel
+                are set up, exchange/queue declared, delivery confirmation is enabled, etc.),
+                or raise an exception if some issue is encountered in that process. Default is None
         """
         logger.debug('Starting thread with callback')
-        if callback is not None:
-            self._on_ready_callback = callback  # to be invoked after all pika setup is done
+        if is_ready is not None:
+            self._is_ready_future = is_ready  # to be invoked after all pika setup is done
         self._thread.start()
```

Comment:
I think you should put an if self._thread.is_alive around this, but that is more for the original problem not what is covered in this PR

Reply:
I went back on forth on this, and either way your comment points to how messy and hard-to-read this class has gotten. 

I put an `is_alive()` check in the public method `start()`, which in turn calls this private `_start()` method. However, the methods aren't identical: public start doesn't accept a callback/Future argument, which the private `_start` does. And `_wait_for_channel_to_be_ready` calls the private start, not the public one.

Should I just combine them to be one consistent function? Or is there a better organization?

Reply:
I would put the 'is_alive' check right before the thread.start(), that means it is always checked. As of right now I don't see the need for a public and private start, so I'd get rid of the private, which is how rpc_client.py is written.